### ISSUE TITLE
Crash with PVR

### ISSUE THREAD

**Issue by edenmedia** [peak3d/inputstream.adaptive#1362]
### Describe the bug
Here is a clear and concise description of what the problem is:

Often using Pvr Simple Client or Zattoo PVR Client, Libreelec crashes and starts again on its own. This happens especially when switching from one channel to another 

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Change channels freely without libreelec crashing

## Actual Behavior

Crash 

### To Reproduce

1. in Zattoo Pvr or Smple Client changes channels repeatedly and after a while libreelec will crash
2.
3.

### Debuglog

The log will comming soon
Here the log https://paste.kodi.tv/agixuronal.kodi
Here the crash log https://paste.kodi.tv/usokuceyoj.kodi
**New LOG here:** https://paste.kodi.tv/togugibati.kodi
**New Crash Log here:** https://paste.kodi.tv/uzeditiqug.kodi

**Logs with the test version of inputstream.adaptive Nr. 2**
**Log: https://paste.kodi.tv/cayevowama.kodi
Crash Log: https://paste.kodi.tv/mikeduyiri.kodi**

### MPD/M3U8s/ISM

An example or copy of a manifest (or manifests for HLS - master and variants) can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ X] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Librerelec
 - Kodi version: Libreelec 11.03
 - inputstream.adaptive version: latest

*note: Once the issue is made we require you to update it with new information should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by CastagnaIT** [comment]
in awaiting for a debug log,
would be interesting also to know if the problem exists also on Kodi 21
since on Kodi 21 there are a lot of fixes/improvements

**Comment by edenmedia** [comment]
Hi, as you suggested I also tried with Libreelec Omega Nightly of 2023-08-21 and unfortunately it continues to crash. I will install Nexus again and post the LOG.
I tried to ask also colleagues who use Libreelec Nexus and all confirm the problem of crash with PVR

**Comment by edenmedia** [comment]
Hi, as I said :
here the log https://paste.kodi.tv/agixuronal.kodi
here the crash log https://paste.kodi.tv/usokuceyoj.kodi

**Comment by edenmedia** [comment]
Sometimes the system crashes only after you change 2 or 3 channels.

**Debug Loggin enable**
**New LOG here:** https://paste.kodi.tv/togugibati.kodi
**New Crash Log here:** https://paste.kodi.tv/uzeditiqug.kodi

**Comment by CastagnaIT** [comment]
thanks for logs
seem that when ISA comes is being terminated try to make manifest updates
this could be one reason for the crashes

i made a test PR #1363
where some test builds are here: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-1363/2/artifacts

unfurnately i cant provide you the test build for libreelec
maybe @popcornmix can help build ISA test PR for libreelec Nexus? or ping some LE dev here that can do it?

**Comment by edenmedia** [comment]
It would be great to have a test build to try on Libreelec 11.03 hopefully @popcornmix can help us in this regard

**Comment by irontes** [comment]
A new issue has been opened on Inputstream GitHub whose is much more detailed so I think it is better to refer to this one https://github.com/xbmc/inputstream.adaptive/issues/1362

**Comment by heitbaum** [comment]
> It would be great to have a test build to try on Libreelec 11.03 hopefully @popcornmix can help us in this regard

@edenmedia - https://github.com/heitbaum/LibreELEC.tv/blob/isa/inputstream.adaptive-20.3.11.1.zip

**Comment by edenmedia** [comment]
@heitbaum  Thanks a million for your help.

Unfortunately the test version that heitbaum posted does not solve the crash problem. After a few channels 
Libreelec crashed and then started again on its own.

**Here the LOGS with the inputstream version that heitbaum posted** 
**Log: https://paste.kodi.tv/cihidebujo.kodi
Crash Log: https://paste.kodi.tv/ukuvowabet.kodi**

sorry those I had before posted were the old logs with official inputstream I corrected the links and now are the logs with the test version of inputstream

**Comment by CastagnaIT** [comment]
seem i have forgot add a linecode

@heitbaum thank you, have you patience for another round of build?
https://patch-diff.githubusercontent.com/raw/xbmc/inputstream.adaptive/pull/1363.patch

**Comment by heitbaum** [comment]
> seem i have forgot add a linecode
> 
> @heitbaum thank you, have you patience for another round of build? https://patch-diff.githubusercontent.com/raw/xbmc/inputstream.adaptive/pull/1363.patch

@CastagnaIT  / @edenmedia - re compiled - tagged as test2 - same place.

**Comment by CastagnaIT** [comment]
thanks
@edenmedia download by using same link
https://github.com/heitbaum/LibreELEC.tv/blob/isa/inputstream.adaptive-20.3.11.1.zip

**Comment by edenmedia** [comment]
ok thanks I try it right now

**Comment by edenmedia** [comment]
Ok I tried just now unfortunately after changing some channels Libreelec crashed. However, it seems to last longer than the previous version but it can only be a coincidence .

**Nr.2 Log: https://paste.kodi.tv/cayevowama.kodi
Nr.2 Crash Log: https://paste.kodi.tv/mikeduyiri.kodi**

**Comment by CastagnaIT** [comment]
unfurnately i have not found a way to test locally so no idea where the root of problem or if i have done some mistake on my previous attempts
i tried make new changes, hoping better, try and check thats is not so convenient way

if @heitbaum have time to build it again
https://github.com/xbmc/inputstream.adaptive/pull/1363.patch

**Comment by heitbaum** [comment]
> unfurnately i have not found a way to test locally so no idea where the root of problem or if i have done some mistake on my previous attempts i tried make new changes, hoping better, try and check thats is not so convenient way
> 
> if @heitbaum have time to build it again https://github.com/xbmc/inputstream.adaptive/pull/1363.patch

Built and pushed. https://github.com/heitbaum/LibreELEC.tv/blob/isa/inputstream.adaptive-20.3.11.1.zip

**Comment by edenmedia** [comment]
Unfortunately this one didn't work either, Libreelec keeps crashing.  I really do appreciate all the work everyone is putting in to resolve this issue.  Do not worry @CastagnaIT, I really do not mind testing out this modifications (I am more than happy to test it and try to contribute to resolving this). 

**Nr.3 Log:  https://paste.kodi.tv/toyirodumu.kodi
Nr.3 Crash Log:  https://paste.kodi.tv/ozufuxikow.kodi**

I don't know if this can help, here is the new log generated after the crash:
**Nr.3 New Log After Crash -> https://paste.kodi.tv/jutawoguvu.kodi**

**Comment by CastagnaIT** [comment]
@heitbaum can you rebuild? thank you
https://github.com/xbmc/inputstream.adaptive/pull/1363.patch

**Comment by dobo90** [comment]
> Unfortunately this one didn't work either, Libreelec keeps crashing. I really do appreciate all the work everyone is putting in to resolve this issue. Do not worry @CastagnaIT, I really do not mind testing out this modifications (I am more than happy to test it and try to contribute to resolving this).
> 
> **Nr.3 Log: https://paste.kodi.tv/toyirodumu.kodi Nr.3 Crash Log: https://paste.kodi.tv/ozufuxikow.kodi**
> 
> I don't know if this can help, here is the new log generated after the crash: **Nr.3 New Log After Crash -> https://paste.kodi.tv/jutawoguvu.kodi**

```
Thread 1 (Thread 0x7fd9a57fe6c0 (LWP 1100)):
#0  0x00007fda47eb3f0c in ?? () from /usr/lib/libc.so.6
#1  0x00007fda47e68162 in raise () from /usr/lib/libc.so.6
#2  0x00007fda47e54464 in abort () from /usr/lib/libc.so.6
#3  0x00007fda47ea8c08 in ?? () from /usr/lib/libc.so.6
#4  0x00007fda47ebca4a in ?? () from /usr/lib/libc.so.6
#5  0x00007fda47ec08cc in malloc () from /usr/lib/libc.so.6
#6  0x00007fda47cc14fc in operator new(unsigned long) () from /usr/lib/libstdc++.so.6
#7  0x00007fda47d57b8a in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_assign(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /usr/lib/libstdc++.so.6
#8  0x000000000115bb2a in CFileItem::operator=(CFileItem const&) ()
#9  0x000000000115c8b2 in CFileItem::CFileItem(CFileItem const&) ()
#10 0x0000000000c47030 in CVideoPlayer::Prepare() ()
#11 0x0000000000c490ce in CVideoPlayer::HandleMessages() ()
#12 0x0000000000c4a0ca in CVideoPlayer::Process() ()
#13 0x0000000000e52843 in CThread::Action() ()
#14 0x00000000016be613 in ?? ()
#15 0x0000000000e4762f in ?? ()
#16 0x00007fda47cec403 in ?? () from /usr/lib/libstdc++.so.6
#17 0x00007fda47eb2460 in ?? () from /usr/lib/libc.so.6
#18 0x00007fda47f2ba2c in ?? () from /usr/lib/libc.so.6
```

This seems to be a duplicate of https://github.com/xbmc/xbmc/issues/23247.

**Comment by heitbaum** [comment]
> @heitbaum can you rebuild? thank you https://github.com/xbmc/inputstream.adaptive/pull/1363.patch

Test4 compiled. Same place. https://github.com/heitbaum/LibreELEC.tv/blob/isa/inputstream.adaptive-20.3.11.1.zip  Here is the commit.

https://github.com/heitbaum/LibreELEC.tv/commit/ac1b51805efad54e9de62a44b55535cf94781555

**Comment by edenmedia** [comment]
Unfortunately Test4 also did not work, Libreelec still crashes.

**Here are the logs:**
**Nr.4 Log: https://paste.kodi.tv/unilafufiv.kodi
Nr.4 Crash Log: https://paste.kodi.tv/agigarefab.kodi**

_EDIT:_
_I also tried Test4 on an other device and the Logs look different so I updated these logs just in case they were needed._
**Nr.4_1 Log: https://paste.kodi.tv/vehanoqaco.kodi
Nr.4_1 Crash Log: https://paste.kodi.tv/etefuyorav.kodi**

**Comment by CastagnaIT** [comment]
yes i noticed that the crash now is on Kodi side not from this addon
but last test was to make sure that RefreshLiveSegments is not executed (after ~CSession)
and strangely it keeps starting up where it shouldn't...

@dobo90 could be same but you should update logs on your Issue https://github.com/xbmc/xbmc/issues/23247
because they no longer exist then cannot be compared

**Comment by fritsch** [comment]
Related: https://github.com/xbmc/xbmc/issues/23247 ?

**Comment by CastagnaIT** [comment]
seem so at least as @dobo90 reported, unfurnately the original logs https://github.com/xbmc/xbmc/issues/23247 are expired
he has added other logs, and also proposed a code change on https://github.com/xbmc/xbmc/issues/23247 description which could be a solution

there is to say that in this Issue use case, there is an additional bug
that must be fixed by #1363 in order to catch the kodi side bug, that can be seen on log of comment https://github.com/xbmc/inputstream.adaptive/issues/1362#issuecomment-1696548613

here we have also another similar bug in the issue #1185 that concern always when you switch playback by PVR
but from log the error seem a bit different

**Comment by edenmedia** [comment]
> seem so at least as @dobo90 reported, unfurnately the original logs [xbmc/xbmc#23247](https://github.com/xbmc/xbmc/issues/23247) are expired he has added other logs, and also proposed a code change on [xbmc/xbmc#23247](https://github.com/xbmc/xbmc/issues/23247) description which could be a solution
> 
> there is to say that in this Issue use case, there is an additional bug that must be fixed by #1363 in order to catch the kodi side bug, that can be seen on log of comment [#1362 (comment)](https://github.com/xbmc/inputstream.adaptive/issues/1362#issuecomment-1696548613)
> 
> here we have also another similar bug in the issue #1185 that concern always when you switch playback by PVR but from log the error seem a bit different

What does this mean that the problem is Libreelec and not inputstream? In this case, do we have to wait for a new release of Libreelec?

**Comment by CastagnaIT** [comment]
@edenmedia as we can see there are more than one bug
for now too early to speak about,

to be sure if @dobo90 fix works also for this use case means build libreelec v20 with
https://github.com/xbmc/xbmc/issues/23247 proposed patch
and use it in conjunction with last ISA test build attached here

ofc the bug on Kodi side means that each other client like LibreElec will have to update their sources in order to have the new bugfix, then end users will have to wait a new LE version, this will require many time but use nightly builds can be more fast

**Comment by edenmedia** [comment]
So in other words, the fastest way of hopefully solving this issue is by asking the libreelec team to build a new **Generic Legacy Nightly** build of "Nexus" in which to include the fix (https://github.com/xbmc/xbmc/issues/23247) and then try again by installing the Test Nr.4?

**Comment by CastagnaIT** [comment]
atm i have no idea if will be backported to Nexus, so for now we have to wait
when the fix will be merged in to official kodi
we have to check what LE nightly have the fix, that should be downlaoded from here https://test.libreelec.tv/

**Comment by edenmedia** [comment]
@CastagnaIT Hello thanks for the info but I had already tried to download the latest LE nightly unfortunately even in the latest versions the problem of the crash has not yet been solved both in versions 11 (Nexus) and 12 (Omega).
The biggest problem is that libreelec Nexus is almost useless (at least for those who use it mainly for PVR) without a solution to these Crashes, and downgrade to the Matrix is impossible since many addons have stopped giving support to versions prior to the Nexus (example Zattoo Pvr)

@heitbaum I think the only one who can help us is you making a last effort to build a version of Libreelec Generic Legacy 11.3 (Nexus) with integrated the solution by @dobo90  https://github.com/xbmc/xbmc/issues/23247.
I'm sure many users will worship you as a saint :)

**Comment by CastagnaIT** [comment]
you have not understand nothing...
the fix on kodi sources has not be approved yet
you will have to wait a week or more weeks or months,
then after this, you will have to wait that LE align his sourcecode to kodi and make new test builds,
at today there is nothing

**Comment by edenmedia** [comment]
@CastagnaIT I suspect you didn't understand nothing! What I wanted to tell you and that I had already checked in the nightly releases and I saw that there were no references to our problem, Since I am convinced that it will take a long time for an official solution, I asked  @heitbaum  if he could make an effort in compiling a Test version of libreelec with @dobo90 solution in the meantime that a new official release comes out.  I also asked the developer of Zattoo PVR https://github.com/rbuehlma/pvr.zattoo/issues/185 if he could at least upgrade his addon for the Matrix version so at least users can fix the problem by downgrading to Libreelec 10.
Right now Libreelec Nexus and Omega are useless for those who use the PVR so I'm looking for a solution otherwise what can be done? Leave Libreelec?

**Comment by CastagnaIT** [comment]
> I suspect you didn't understand nothing! What I wanted to tell you and that I had already checked in the nightly releases and I saw that there were no references to our problem

No wonder they are still working on the fix PR,
until there is a final approved fix, there is no point in bothering outsider people with these requests

**Comment by edenmedia** [comment]
> > I suspect you didn't understand nothing! What I wanted to tell you and that I had already checked in the nightly releases and I saw that there were no references to our problem
> 
> No wonder they are still working on the fix PR, until there is a final approved fix, there is no point in bothering outsider people with these requests

**Comment by edenmedia** [comment]
> > > I suspect you didn't understand nothing! What I wanted to tell you and that I had already checked in the nightly releases and I saw that there were no references to our problem
> > 
> > 
> > No wonder they are still working on the fix PR, until there is a final approved fix, there is no point in bothering outsider people with these requests
ok

**Comment by edenmedia** [comment]
I'm just trying to find a temporary solution because I believe an official solution will take a long time.

**Comment by CastagnaIT** [comment]
i found a way to reproduce your crashes locally
so i made the definitive fixes PR's for ISA Omega/Nexus

after ISA fixes will be merged this Issue will be so closed because kodi bugs are not related to this component,
but dont worry, i suggest you to follow the kodi PR https://github.com/xbmc/xbmc/pull/23710
to know when will be merged, after the merge you need to wait the new LE nightly build that usually should happen with regularity
